test-data/unit/check-modules.test

```diff
@@ -811,3 +811,35 @@ from a import x
 [file a/__init__.py]
 x = 0
 [out]
+
+
+-- Test stability under import cycles
+-- ----------------------------------
+
+-- The two tests are identical except one main has 'import x' and the other 'import y'
+
+[case testImportCycleStability1]
+import x
+[file x.py]
+class Base:
+    attr = 'x'
```

Comment:
Maybe use a trickier initializer here so that better type inference is less likely to make the order irrelevant.